### DISCUSSION TITLE
Zenodo releases and Firedrake versioning

### DISCUSSION THREAD

**Discussion by miguelcoolchips** [firedrakeproject/firedrake#2930] [General]
As far as I know, the best way to consistently repeat results using Firedrake is creating a Zenodo release and passing its corresponding DOI to the `firedrake-install` script. However, I believe there are two potential issues with this approach: 

One is that the `firedrake-install` script itself is subject to changes and the installation could crash. This problem can be solved if one downloads the `firedrake-install` script in the Firedrake commit given by the Zenodo release. 

The other issue is that the packages installed by the OS package manager are not pinned to any versions, which can create problems, as we saw in #2793. Perhaps one solves this with the approach suggested by David Ham in that issue.

Are there other problems I am not aware of? Thanks.

**Comment by dham** [comment]
Firedrake-zenodo will get you back our software, but it doesn't preserve the environment. If you want full environment preservation for the hard end of reproducibility then you probably need to work in a container and archive that container.
